tools/document-pii/src/main.rs

```diff
@@ -0,0 +1,650 @@
+    match item {
+        Some(path) => {
+            if let Some(structorenum) = {
+                let guard = ALL_TYPES.lock().unwrap();
+                guard.get(path).cloned()
+            } {
+                let module_path = path.rsplit_once("::").unwrap().0.to_owned();
+                let theitem = structorenum;
+                let mut visitor = TypeVisitor {
+                    module_path,
+                    ..Default::default()
+                };
+                match theitem {
+                    EnumOrStruct::Struct(itemstruct) => visitor.visit_item_struct(&itemstruct),
+                    EnumOrStruct::Enum(itemenum) => visitor.visit_item_enum(&itemenum),
+                };
+            } else {
+                panic!("Please provide a fully qualified path to a struct or an enum. E.g. 'relay_general::protocol::Event'");
+            }
+        }
+        None => {
+            let all_types = ALL_TYPES.lock().unwrap().clone();
+            for (key, value) in all_types.iter() {
+                let module_path = key.rsplit_once("::").unwrap().0.to_owned();
+                let mut visitor = TypeVisitor {
+                    module_path,
+                    ..Default::default()
+                };
+                match value {
+                    EnumOrStruct::Struct(itemstruct) => visitor.visit_item_struct(itemstruct),
+                    EnumOrStruct::Enum(itemenum) => visitor.visit_item_enum(itemenum),
+                };
+            }
+        }
+    }
+}
+
+fn get_pretty_pii_field_format(unnamed_replace: String) -> Vec<Pii> {
+    let mut output_vec = vec![];
+    for pii in PII_TYPES.lock().unwrap().iter() {
+        let mut output = Pii::default();
+        output.path.push_str(&pii[0].qualified_type_name);
+
+        for path in pii {
+            output.path.push_str(&format!(".{}", path.field_name));
+        }
+
+        output.path = output.path.replace("{{Unnamed}}.", &unnamed_replace);
+        output_vec.push(output);
+    }
+    output_vec.sort_by_key(|pii| pii.path.clone());
+    output_vec
+}
+
+#[derive(Debug, Serialize, Default)]
+struct Pii {
+    path: String,
+    //description: String,
+}
+
+// Due to global variables, have to both run the tests sequentially as well as clearing the globals
+#[cfg(test)]
+mod tests {
+    use std::str::FromStr;
+
+    use serial_test::serial;
+
+    use super::*;
+    const RUST_CRATE: &str = "../../tests/test_pii_docs";
+
+    fn clear_globals() {
+        ALL_TYPES.lock().unwrap().clear();
+        USE_STATEMENTS.lock().unwrap().clear();
+        PII_TYPES.lock().unwrap().clear();
+        CURRENT_PATH.lock().unwrap().clear();
+    }
+
+    #[test]
+    #[serial]
+    fn test_use_statements() {
+        clear_globals();
+        let rust_crate = PathBuf::from_str(&RUST_CRATE).unwrap();
+        run(&rust_crate, &vec![], None);
+        let use_statements = USE_STATEMENTS.lock().unwrap().clone();
+        insta::assert_debug_snapshot!(use_statements);
```

Comment:
while the use_statements is an intermediate variable, its not "instead" because we also test the final output, I guess you could make a case that its redundant but at least like this if something breaks then it makes it easier to debug, whats your thoughts on that?

Reply:
true but there already is code that directly tests the output, this use_statements test is still helpful if i want to modify the code dealing with use_statements

Reply:
I may have missed it, but what test is testing the output? The output should be something like it used to be in the PR description.

Reply:
`test_pii_true`, `test_pii_false` and `test_pii_all` are testing the output, if you look at the snapshot files its how the files that get sent to the sentry docs will look like